cmd/integrationArtifactTransport.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return err
+	}
+	return CreateIntegrationArtifactTransportRequest(config, apimData)
+}
+
+//CreateIntegrationArtifactTransportRequest - Create a transport request for Integration Package
+func CreateIntegrationArtifactTransportRequest(config *integrationArtifactTransportOptions, apistruct apim.Bundle) error {
+	httpMethod := http.MethodPost
+	httpClient := apistruct.Client
+	createTransportRequestURL := fmt.Sprintf("%s/v1/contentResources/export", apistruct.Host)
+	header := make(http.Header)
+	header.Add("content-type", "application/json")
+	payload, jsonError := GetCPITransportReqPayload(config)
+	if jsonError != nil {
+		return errors.Wrapf(jsonError, "Failed to get json payload for file %v, failed with error", config.IntegrationPackageID)
+	}
+
+	createTransportRequestResp, httpErr := httpClient.SendRequest(httpMethod, createTransportRequestURL, payload, header, nil)
+
+	if httpErr != nil {
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", httpMethod, createTransportRequestURL)
+	}
+
+	if createTransportRequestResp != nil && createTransportRequestResp.Body != nil {
+		defer createTransportRequestResp.Body.Close()
+	}
+
+	if createTransportRequestResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response")
+	}
+
+	if createTransportRequestResp.StatusCode == http.StatusAccepted {
+		log.Entry().
+			WithField("IntegrationPackageID", config.IntegrationPackageID).
+			Info("successfully created the integration package transport request")
+
+		bodyText, readErr := ioutil.ReadAll(createTransportRequestResp.Body)
+		if readErr != nil {
+			return errors.Wrap(readErr, "HTTP response body could not be read")
+		}
+		jsonResponse, parsingErr := gabs.ParseJSON([]byte(bodyText))
+		if parsingErr != nil {
+			return errors.Wrapf(parsingErr, "HTTP response body could not be parsed as JSON: %v", string(bodyText))
+		}
+		processId := jsonResponse.Path("processId").Data().(string)
+
+		if processId != "" {
+			error := pollTransportStatus(processId, retryCount, config, httpClient, apistruct.Host)
+			return error
+		}
+		return errors.New("Invalid process id")
+	}
+	responseBody, readErr := ioutil.ReadAll(createTransportRequestResp.Body)
+
+	if readErr != nil {
+		return errors.Wrapf(readErr, "HTTP response body could not be read, response status code: %v", createTransportRequestResp.StatusCode)
+	}
+	log.Entry().Errorf("a HTTP error occurred! Response body: %v, Response status code : %v", string(responseBody), createTransportRequestResp.StatusCode)
+	return errors.Errorf("integration flow deployment failed, response Status code: %v", createTransportRequestResp.StatusCode)
+}
+
+//pollTransportStatus - Poll the integration package transport processing, return status or error details
+func pollTransportStatus(processId string, retryCount int, config *integrationArtifactTransportOptions, httpClient piperhttp.Sender, apiHost string) error {
+
+	if retryCount <= 0 {
```

Comment:
Wouldn't `retryCount <= 0` mean that it didn't retry? The error message says "after retrying several times"

Reply:
retryCount  getting decremented in the recursive function call. once its become 0, then its clear after several attempts we couldn't get the status

Reply:
Ah okay. Maybe you could change the variable name to something like `remainingRetries` so it's more clear when just looking at the part of the code?

Reply:
done